staging/src/k8s.io/apimachinery/pkg/apis/config/v1alpha1/defaults.go

```diff
@@ -16,15 +16,7 @@ limitations under the License.
 
 package v1alpha1
 
-import (
-	kruntime "k8s.io/apimachinery/pkg/runtime"
-)
-
-func addDefaultingFuncs(scheme *kruntime.Scheme) error {
-	return RegisterDefaults(scheme)
-}
-
-func SetDefaults_ClientConnectionConfiguration(obj *ClientConnectionConfiguration) {
+func DefaultClientConnectionConfiguration(obj *ClientConnectionConfiguration) {
```

Comment:
`RecommendedDefaultClientConnectionConfiguration`. godoc indicating that it is subject to change as our recommendations change (same as the options structs we have today in apiserver) and an explanation that providing this method is a courtesy if you find that wish to use it in your config.  It is intentionallly not registered in the scheme so that callers have flexibility to choose their own defaults.  Conversion relies on defaulting, so if we did not do this, our conversion may accidentally rely on this.

Reply:
added godoc comment to both types